src/_pytest/mark/structures.py

```diff
@@ -37,6 +42,8 @@ def get_empty_parameterset_mark(config, argnames, func):
         mark = MARK_GEN.skip
     elif requested_mark == "xfail":
         mark = MARK_GEN.xfail(run=False)
+    elif requested_mark == "fail_at_parametrize":
+        raise ParametrizeError("Empty parameter set")
```

Comment:
please include the function name in the error just to ease the ux

Reply:
Do you mean "get_empty_parameterset_mark" in the error message? I was actually thinking that it would be nice to have some introspection providing the name of the test which generated the error, but have no clue if/how that can be done.

Reply:
no, i meant the name of the function object - so its more clear where the missmark happened

Reply:
Fixed and added test for it.